sdk/storage/azure-storage-file-datalake/src/test/java/com/azure/storage/file/datalake/FileAPITest.groovy

```diff
@@ -1021,6 +1026,112 @@ class FileAPITest extends APISpec {
         thrown(NullPointerException)
     }
 
+    def "Download to file exists"() {
+        setup:
+        def testFile = new File(testName + ".txt")
+        if (!testFile.exists()) {
+            assert testFile.createNewFile()
+        }
+        fc.append(new ByteArrayInputStream(defaultData.array()), 0, defaultDataSize)
+        fc.flush(defaultDataSize)
+
+        when:
+        // Default overwrite is false so this should fail
+        fc.downloadToFile(testFile.getPath())
+
+        then:
+        def ex = thrown(UncheckedIOException)
+        ex.getCause() instanceof FileAlreadyExistsException
+
+        cleanup:
+        testFile.delete()
+    }
+
+    def "Download to file exists succeeds"() {
+        setup:
+        def testFile = new File(testName + ".txt")
+        if (!testFile.exists()) {
+            assert testFile.createNewFile()
+        }
+        fc.append(new ByteArrayInputStream(defaultData.array()), 0, defaultDataSize)
+        fc.flush(defaultDataSize)
+
+        when:
+        fc.downloadToFile(testFile.getPath(), true)
+
+        then:
+        new String(Files.readAllBytes(testFile.toPath()), StandardCharsets.UTF_8) == defaultText
```

Comment:
Can we converge on using the compare files method we have for blobs? It might actually be worthwhile to have a test common at some point... 

And testing a few different data sizes and different values for the ParallelTransferOptions would be good, too

Reply:
Will do, the only reason I didnt port those tests over was because I havent ported the uploadFromFile methods yet and I wanted to wait til I had that to work with larger datasets

Reply:
If you're planning on adding those methods and tests before the next release, I'm fine on waiting for the next PR

Reply:
just added the extra download to file tests